service/matching/task_gc.go

```diff
@@ -33,11 +33,13 @@ import (
 )
 
 type taskGC struct {
+	backlogMgr *backlogManagerImpl
```

Comment:
Hm, this (and the others) make the cyclical dependency situation worse. Are we going to declare design bankruptcy on these components and merge them all into one place in the near future?

Reply:
Oh, funny, on the tip of my working branch I switched it back to taking separate dependencies... I extracted this PR from a few in the middle and forgot about that. I'll change it to use separates now.

I have definitely considered merging them and still think it may make sense. It's partly a question of how much you want to review as "diff" and how much as new code.

Inherently there is always going to be some cycle somewhere since I think backlog manager <--> task reader is irreducible. We can hide it inside closures if we want, but I feel like that can obscure more than it helps.

Reply:
Sounds good!

I'd support merging them as an immediate step since they are too entangled right now anyway. Then, it might be possible to chip off parts again one-by-one to reduce the size where reasonable/possible.

I think the major downsides of the cycles are (1) testing (interfaces help, though) and (2) reasoning about it. A reset might help identify the parts that can actually be safely extracted. Only where needed/valuable ofc.